### PR TITLE
Allow the duplicate set comparator to be set.

### PR DESCRIPTION
**Pull request by nh13**
@tfenne @yfarjoun simple change to allow a user-provided comparator, for example, if we wanted to extend `SAMRecordDuplicateComparator` with our own implementation.

### CODE REVIEW

**Comment by eitanbanks** [comment]
@nh13 wouldn't you want to add this to the constructor?  There's already a DuplicateSet being generated there with the default comparator.

**Comment by nh13** [comment]
@eitanbanks I think the proliferation of alternate constructors would start to make this difficult to understand.  If you would prefer, I can move it to more of a builder pattern?   I can have `setComparator` and `setScoringStrategy` return a `DuplicateSetIterator`?   I could go further and replace the constructors to make it truly a builder pattern but that would make it not backwards compatible.  Your call for all of this, I just need to set the comparator for my own purposes.

**Comment by eitanbanks** [comment]
Sorry, I probably wasn't clear.  I just meant that if you don't do something then your results might not be accurate.  The constructor already has this in it:
this.duplicateSet = new DuplicateSet(this.comparator);
So if you wait until after the constructor to set your comparator then you will have missed at least one creation of a duplicate said with the correct comparator.

**Comment by nh13** [comment]
@eitanbanks I suppose then `setScoringStrategy` is also being set after the first duplicate set is populated.  I think that affects Picard's `SimpleMarkDuplicatesWithMateCigar`.  What do you think about making `setScoringStrategy` to deprecated, since now folks can set the scoring strategy on the passed in comparator?

File: src/java/htsjdk/samtools/DuplicateSetIterator.java [hunk 1]
```diff
@@ -85,6 +94,8 @@ public DuplicateSetIterator(final CloseableIterator<SAMRecord> iterator, final S
 
     }
 
+    @Deprecated
+    /** Do not use as the first duplicate set will not be compared with this scoring strategy. */
```
Comment (eitanbanks) [inline src/java/htsjdk/samtools/DuplicateSetIterator.java]: "Instead, set the scoring strategy on the comparator"

**Comment by eitanbanks** [comment]
Yeah, this looks better

**Comment by nh13** [comment]
@eitanbanks thanks for the review, I will merge once the tests pass.

**Pull request merged**